frontend/src/views/LevelView.vue

```diff
@@ -0,0 +1,183 @@
+          <h5 style="padding-bottom: 5px">Link to Knowledge Path</h5>
+          <ElInput
+            style="width: 417px; height: 56px"
+            placeholder="Link"
+            v-model="levelLink"
+          ></ElInput>
+        </div>
+      </div>
+      <div class="SaveButton">
+        <ElButton
+          type="primary"
+          style="width: 215px; height: 32; border-radius: 4px"
+          @click="CreateLevel()"
+        >
+          Save
+        </ElButton>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { Role } from '@/domain/Role'
+import type { Level } from '@/domain/Level'
+import { useRolesStore } from '@/stores/roles'
+import { ref, onMounted, computed } from 'vue'
+import { ElInput, ElButton } from 'element-plus'
+import { ErrorMessage, SuccessMessage } from '@/services/messages'
+import { useRouter } from 'vue-router'
+import { HttpServerError } from '@/services/http'
+
+const roleStore = useRolesStore()
+const roleId = ref<string>('1')
+const role = ref<Role>()
+const levelName = ref<string>('')
+const levelMinRange = ref<string>('')
+const levelMaxRange = ref<string>('')
+const levelLink = ref<string>('')
+const newLevel = ref<Level>({ name: '', minCoeficient: 0, maxCoeficient: 1, link: '' })
+const router = useRouter()
+const title = ref<string>('New Level')
+const newLevelTitle = computed(() => {
+  return `${role.value?.name || ''} - ${levelName.value || title.value}`
+})
+
+onMounted(async () => {
+  role.value = await roleStore.getRoleById(roleId.value)
+})
+
+const RedirectToRoleView = () => {
+  router.push('/role-view')
+  SuccessMessage('Your level was created')
+}
+
+const CreateLevel = async () => {
+  if (
+    levelName.value == '' ||
+    levelMinRange.value == '' ||
+    levelMaxRange.value == '' ||
+    levelLink.value == ''
+  ) {
+    ErrorMessage('You need to fill all the fields')
+    return
+  }
+  try {
+    if (role.value) {
+      if (parseInt(levelMinRange.value) >= 0 && parseInt(levelMaxRange.value) >= 0) {
+        newLevel.value.name = levelName.value
+        newLevel.value.minCoeficient = parseInt(levelMinRange.value)
+        newLevel.value.maxCoeficient = parseInt(levelMaxRange.value)
+        newLevel.value.link = levelLink.value
+        RedirectToRoleView()
+        await roleStore.addLevel(newLevel.value, role.value)
+      } else {
+        ErrorMessage('MinRange and MaxRange should be positive numbers')
+      }
+    }
+  } catch (error: any) {
+    if (error.status === HttpServerError.HTTP_STATUS_CODE_CONFLICT) {
+      ErrorMessage('This level name already exists')
+    }
+  }
+}
+</script>
+
+<style scoped lang="scss">
+.PageWrapper {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.SaveButton {
+  margin-top: 50px;
+}
+
+.FlexContainer {
+  display: flex;
+  justify-content: space-between;
+}
+
+.MaxAndMinInputs,
+.LinkSide {
+  flex: 1;
+}
+
+.LinkSide {
+  --el-border-radius-base: 10px;
```

Comment:
Variables should not be declared here. This is a spacing variable. Create a file theme/spacing.scss and put the spacing variables there